web/src/test/java/com/graphhopper/http/resources/CustomWeightingRouteResourceTest.java

```diff
@@ -0,0 +1,129 @@
+        GraphHopperServerConfiguration config = new GraphHopperServerTestConfiguration();
+        config.getGraphHopperConfiguration().
+                putObject("graph.flag_encoders", "bike,car").
+                putObject("routing.ch.disabling_allowed", true).
+                putObject("prepare.min_network_size", 0).
+                putObject("prepare.min_one_way_network_size", 0).
+                putObject("datareader.file", "../core/files/north-bayreuth.osm.gz").
+                putObject("graph.location", DIR).
+                putObject("graph.encoded_values", "max_height,max_weight,max_width,hazmat,toll,surface,track_type").
+                setProfiles(Arrays.asList(
+                        new CustomProfileConfig("car").setCustomModel(new CustomModel()).setVehicle("car"),
+                        new CustomProfileConfig("bike").setCustomModel(new CustomModel()).setVehicle("bike"),
+                        new CustomProfileConfig("truck").setVehicle("car").
+                                putHint("custom_model_file", "./src/test/resources/com/graphhopper/http/resources/truck.yml"),
+                        new CustomProfileConfig("cargo_bike").setVehicle("bike").
+                                putHint("custom_model_file", "./src/test/resources/com/graphhopper/http/resources/cargo_bike.yml"))).
+                setCHProfiles(Collections.singletonList(new CHProfileConfig("truck")));
+        return config;
+    }
+
+    @BeforeAll
+    @AfterAll
+    public static void cleanUp() {
+        Helper.removeDir(new File(DIR));
+    }
+
+    @Test
+    public void testCHTruckQuery() {
+        String jsonQuery = "{" +
+                " \"points\": [[11.58199, 50.0141], [11.5865, 50.0095]]," +
+                " \"profile\": \"truck\"" +
+                "}";
+        final Response response = clientTarget(app, "/route").request().post(Entity.json(jsonQuery));
+        assertEquals(200, response.getStatus());
+        JsonNode json = response.readEntity(JsonNode.class);
+        JsonNode infoJson = json.get("info");
+        assertFalse(infoJson.has("errors"));
+        JsonNode path = json.get("paths").get(0);
+        assertEquals(path.get("distance").asDouble(), 1500, 10);
+        assertEquals(path.get("time").asLong(), 151_000, 1_000);
+    }
+
+    @Test
+    public void testAvoidArea() {
+        String yamlQuery = "points: [[11.58199, 50.0141], [11.5865, 50.0095]]\n" +
+                "profile: car\n";
+
+        JsonNode yamlNode = clientTarget(app, "/custom").request().post(Entity.entity(yamlQuery,
+                new MediaType("application", "yaml"))).readEntity(JsonNode.class);
+        JsonNode path = yamlNode.get("paths").get(0);
+        assertEquals(path.get("distance").asDouble(), 661, 10);
+
+        yamlQuery += "priority:\n" +
+                // todonow: shall we increase the priority here to get the route that crosses the polygon, but uses
+                // a faster road (see #2021)? or maybe do both?
```

Comment:
?

Reply:
Yes, probably both and link to the issue